q/verbose/verbose.q

```diff
@@ -0,0 +1,471 @@
+//@param t the input table (has to be keyed and the keys will be the new columns, original columns will be the new keys)
+//@param newColName  name of the key column in the result table
+//@return the transposed table
+.finos.verbose.transposeGEN: {[t; newColName]
+    if[0 = count t; '"Transpose with empty table"];
+
+    //works on single columnkeyed table
+    newCols: first value flip key t;
+    transposedData : flip value flip value t;
+    flip[enlist[newColName]!enlist cols value t]! flip [newCols!transposedData]
+ };
+
+///
+//Projection of the .finos.verbose.transposeGEN keeping the key column name in the result table
+//@param t the input table (has to be keyed and the keys will be the new columns, original columns will be the new keys)
+//@return the transposed table
+.finos.verbose.transpose:{[t] .finos.verbose.transposeGEN[t; first keys t] };
+
+///
+// same as .finos.verbose.transpose but it converts the key set to symbols thus can handle e.g. keys of type date or string.
+.finos.verbose.transposeSafe: {[t] .finos.verbose.transpose .finos.verbose.keyToSym t }
+
+////
+// Drops columns that contain null values for all elements
+.finos.verbose.dropFullNullCols: {[t]
+  :(where all each .finos.verbose.safenull flip t) _ t;
+  }
+
+////
+// Drops elements of a dictionary that give back nulls (if value is list, only nulls)
+// string and dict values are excluded from the check
+.finos.verbose.dropNullsFromDict: {[x]
+  xx:((key x) where ({(type x) in 0 99h} each value x))_x;
+  ((key xx) where all each value .finos.verbose.safenull xx)_x
+  }
+
+////
+// Drops rows that contain null values for any key element
+.finos.verbose.dropNullKeyedRows: {
+    delete from x where any .finos.verbose.safenull each flip key x
+ };
+
+////
+// Prefix all non keyed column names with a string - useful for lj
+//@param prefix string
+//@param t keyed table
+.finos.verbose.prefixValueCols: {[prefix;t]
+    $[0=count cols value t; t; ((keys t),`$prefix,/: string cols value t) xcol t]
+ };
+
+///
+// Returns a list of timestamps that uniformly cover the min[x] to max[x] time interval with gran second granularity
+//@param gran the granularity in seconds
+//@param x a list of timestamps
+//@return a list of timestamps that uniformly cover the min[x] to max[x] time interval with gran second granularity
+.finos.verbose.timesplit: {[gran;x] min[x]+`timespan$(gran*`float$`timespan$00:00:01) * til floor reciprocal[gran] * reciprocal[`float$`timespan$00:00:01] *`float$max[x] - min[x] };
+
+//apply fun1 on dates before cut date, fun2 on starting from cut date, with respect to start date and end date
+//fun1 is exclusive on cut date, fun2 includes it
+//both fun1 and fun2 have to be a function with 2 date arguments
+.finos.verbose.sliceByDate: {[sd;ed;cd;fun1;fun2] 
+	  $[(sd<cd);fun1[sd;min[ed,cd-1]];()],$[(ed>cd-1);fun2[max[sd,cd];ed];()]
+ };
+
+//by default these are identity, specify the timezone parameter to have these overwritten in setup.q
```

Comment:
Without the referenced setup.q, these are useless as well as anything that depends on them.